### ISSUE TITLE
Add Weights & Biases (wandb) integration for experiment tracking

### ISSUE THREAD

**Issue by christianWissmann85** [christianWissmann85/essence-wars#19]
## Problem

The training scripts import `wandb` but don't actually use it. Proper experiment tracking would help researchers compare runs and share results.

## Current State

- TensorBoard logging exists and works
- `wandb` is in dependencies but unused
- Experiment metadata saved to local JSON files

## Desired Behavior

Optional wandb integration that:
1. Logs training metrics (loss, reward, win rate)
2. Tracks hyperparameters
3. Saves model artifacts
4. Generates comparison dashboards

## Implementation Approach

Add `--wandb` flag to training scripts:

```python
# In ppo.py, alphazero.py, etc.
if args.wandb:
    import wandb
    wandb.init(
        project="essence-wars",
        config=vars(args),
        name=f"{args.observation_mode}_{timestamp}"
    )
    # Log metrics in training loop
    wandb.log({"reward": reward, "loss": loss})
```

## Files to Modify

- `python/scripts/training/ppo.py`
- `python/scripts/training/alphazero.py`
- `python/scripts/training/behavioral_cloning.py`

## Who This Helps

ML researchers who want to track experiments across multiple runs and share results.

## Skills Needed

- Python
- Basic familiarity with wandb or similar tracking tools

**Comment by ZohaibHassan16** [comment]
Hey, can I take on this

**Comment by christianWissmann85** [comment]
Resolved via PR #23 — wandb integration is now merged into master. Thanks @ZohaibHassan16!
